internal/providers/msentraid/msentraid.go

```diff
@@ -215,6 +215,7 @@ func (p Provider) CurrentAuthenticationModesOffered(
 	endpoints map[string]struct{},
 	currentAuthStep int,
 ) ([]string, error) {
+	slog.Debug(fmt.Sprintf("In CurrentAuthenticationModesOffered: sessionMode=%q, supportedAuthModes=%q, tokenExists=%t, providerReachable=%t, endpoints=%q, currentAuthStep=%d\n", sessionMode, supportedAuthModes, tokenExists, providerReachable, endpoints, currentAuthStep))
```

Comment:
Is this a leftover from a debug session? Feels like too much for an actual log message that we would want to show.

Reply:
I did add it during a debug session but I left it on purpose in a [separate commit](b4b505cc049422d03d8a0ecccf1a9eaa3e09ba93) because I think it will be useful in the future:

```
Add debug output
    
These messages should help debug cases where no authentication mode is
offered.
```